buildspec.yml

```diff
@@ -0,0 +1,17 @@
+version: 0.2
+
+phases:
+  build:
+    commands:
+      - ./gradlew build packageLambda
+
+artifacts:
+  files:
+    - 'frontend-service/build/distributions/frontend-service.zip'
+    - 'data-service/build/distributions/data-service.zip'
+    - 'scheduling-manager/build/distributions/scheduling-manager.zip'
+cache:
+  paths:
+    - '/root/.gradle/caches/**/*'
```

Comment:
Ah neat, I was wondering how this would work. Just watch out, some of these files aren't cacheable, and must be deleted before caching. See https://docs.travis-ci.com/user/languages/java/#Caching for details.

Reply:
What I observed for codebuild caching, it basically zip all the files declared and upload to S3.  And download the S3 object and unzip for next build.

Reply:
Right, so the problem is that gradle leaves a lockfile in there, and that could interfere with subsequent gradle runs. We work around this in Travis with this snippet:

https://github.com/blox/blox/blob/653e896c0a056ab2a97210a4a091a1c2a7ee6bc2/.travis.yml#L7-L11

We should do something similar here.

Reply:
I think the intention to remove the lock file and other stuff is not for cleaning the cache for subsequent gradle. As far as I see, without removing those files, build with previous cache file work fine.

The actual propose of removing them is to avoid re-uploading the cache, see (https://github.com/travis-ci/docs-travis-ci-com/issues/358).   Because those lock file will be changed for every build, the build see the change and have to re-upload the whole cache even there is no actual gradle dependency change.

CodeBuild doesn't provide such `before_cache` phase, but I can try it in the `post-build` phase to see if CodeBuild actual compare the cache to avoid re-uploading.

Reply:
Ah, I see, I misunderstood. Thanks for clarifying.